sdk/synapse/Azure.Analytics.Synapse.AccessControl/src/Models/SynapseRoleAssignment.cs

```diff
@@ -41,5 +41,27 @@ internal SynapseRoleAssignment(Guid roleId, Guid principalId, Guid roleDefinitio
                 Id = value.Id,
                 Properties = value.Properties
             });
+
+        public static implicit operator SynapseRoleAssignment(Response response)
+        {
+            switch (response.Status)
+            {
+                case 200:
+                    return DeserializeResponse(response);
+                default:
+
+#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
+                    throw new NotImplementedException();
+#pragma warning restore CA1065 // Do not raise exceptions in unexpected locations
+
+                    //response.Throw();
```

Comment:
If we don't use a `throw` statement here, we get an error that not all paths return a value